src/controller/cve.controller/cve.controller.js

```diff
@@ -38,133 +18,99 @@ async function getCve (req, res) {
     const result = await instance.get('https://cvecps-dev.mitre.org/cps/ws/cve/' + cveId)
 
     // CPS work item was approved
-    if (result) {
-      logger.info(cveId + ' was sent to the user')
+    if (!result.data.Error) {
+      logger.info(cveId + ' was sent to the user.')
       return res.status(200).json(result.data) // send CVE to user
     } else {
       logger.warn('The CPS work item was not approved. ' + cveId + ' was not sent to the user.')
-      return res.status(500).json({ message: 'A problem ocurred. ' + cveId + ' could not be fetched.' }) // send error message to user
+      logger.info(result.data.Error)
+      return res.status(400).json({ message: 'A problem ocurred. ' + cveId + ' could not be fetched.' }) // send error message to user
     }
   } catch (err) {
     logger.error(err.stack)
     return res.status(500).send('Internal Server Error') // send error message to user
   }
 }
 
-// TODO: Request body sanitation
-// MongoDB: Creates a new CVE if it does not exists for the specified CVE id in the CVE's body req. If it exists, it updates the CVE.
+// CPS: Updates a CVE if one exists for the specified CVE ID in the request body
+// MongoDB: Creates a new CVE only if it does not exists for the specified CVE ID in the request body. If it exists, it does not update the CVE.
 async function submitCve (req, res) {
   const cve = req.body
   const cveId = cve.CVE_data_meta.ID
-  const instance = axios.create({
-    timeout: 10000,
-    headers: {
-      'CVE-API-CNA': req.header('CVE-API-CNA'),
-      'CVE-API-SUBMITTER': req.header('CVE-API-SUBMITTER'),
-      'CVE-API-SECRET': req.header('CVE-API-SECRET')
-    }
-  })
 
-  try {
-    const cpsResult = await instance.post('https://cvecps-dev.mitre.org/cps/ws/cve/' + cveId + '/', cve)
-    const updatedCve = JSON.parse(cpsResult.data.Entry)
-
-    // CPS work item was approved
-    if (!cpsResult.data.Error) {
-      logger.info(cveId + ' has been created/updated in CPS')
-      logger.info(cpsResult.data.Message)
-
-      // Store CVE in MongoDB
-      Cve.findOneAndReplace()
-        .byCveId(cveId)
-        .replaceOne({ cve: updatedCve })
-        .setOptions({ upsert: true })
-        .exec((err, result) => {
-          if (err) {
-            logger.warn('An error ocurred in MongoDB. ' + cveId + ' was not created/updated in the database.')
-            logger.error(err.stack)
-          } else {
-            if (result.upserted) {
-              logger.info(cveId + ' was successfully created in MongoDB')
-            } else {
-              logger.info(cveId + ' was successfully updated in MongoDB')
+  // Find CVE in MongoDB
+  Cve.findOne()
+    .byCveId(cveId)
+    .exec((err, result) => {
+      if (err) {
+        logger.warn('An error ocurred in MongoDB while trying to check if ' + cveId + ' already exists.')
+        logger.error(err.stack)
```

Comment:
Include:

return res.status(500).json({ message: 'Failed to create ' + cveId + '.' })